lib/membrane_live_audio_mixer.ex

```diff
@@ -0,0 +1,256 @@
+      ),
+    options: [
+      offset: [
+        spec: Time.non_neg_t(),
+        default: 0,
+        description: "Offset of the input audio at the pad."
+      ]
+    ]
+
+  @impl true
+  def handle_init(_ctx, %__MODULE__{stream_format: stream_format} = options) do
+    if options.native_mixer && !options.prevent_clipping do
+      raise("Invalid element options, for native mixer only clipping preventing one is available")
+    else
+      live_queue =
+        if stream_format == nil,
+          do: nil,
+          else: LiveQueue.init(stream_format)
+
+      state =
+        options
+        |> Map.from_struct()
+        |> Map.put(:mixer_state, initialize_mixer_state(stream_format, options))
+        |> Map.put(:live_queue, live_queue)
+        |> Map.put(:end_of_stream?, false)
+
+      {[], state}
+    end
+  end
+
+  @impl true
+  def handle_playing(_context, %{stream_format: %RawAudio{} = stream_format} = state) do
+    {[stream_format: {:output, stream_format}], state}
+  end
+
+  def handle_playing(_context, %{stream_format: nil} = state) do
+    {[], state}
+  end
+
+  @impl true
+  def handle_stream_format(_pad, stream_format, _context, %{stream_format: nil} = state) do
+    state = %{state | stream_format: stream_format}
+    mixer_state = initialize_mixer_state(stream_format, state)
+    live_queue = LiveQueue.init(stream_format)
+
+    {[stream_format: {:output, stream_format}],
+     %{state | mixer_state: mixer_state, live_queue: live_queue}}
+  end
+
+  @impl true
+  def handle_stream_format(_pad, stream_format, _context, %{stream_format: stream_format} = state) do
+    {[], state}
+  end
+
+  @impl true
+  def handle_stream_format(pad, stream_format, _context, state) do
+    raise(
+      RuntimeError,
+      "received invalid stream_format on pad #{inspect(pad)}, expected: #{inspect(state.stream_format)}, got: #{inspect(stream_format)}"
+    )
+  end
+
+  @impl true
+  def handle_start_of_stream(
+        Pad.ref(:input, pad_id) = pad,
+        context,
+        %{live_queue: live_queue} = state
+      ) do
+    offset = context.pads[pad].options.offset
+
+    new_live_queue = LiveQueue.add_queue(live_queue, pad_id, offset)
+
+    started_input_pads_number =
+      context.pads
+      |> Enum.filter(fn {_id, pad} -> pad.direction == :input and pad.start_of_stream? == true end)
+      |> Enum.count()
+
+    actions =
+      if started_input_pads_number == 1,
+        do: [start_timer: {:initiator, state.latency}],
+        else: []
+
+    {actions, %{state | live_queue: new_live_queue}}
+  end
+
+  @impl true
+  def handle_process(
+        Pad.ref(:input, pad_id),
+        buffer,
+        _context,
+        %{live_queue: live_queue} = state
+      ) do
+    new_live_queue = LiveQueue.add_buffer(live_queue, pad_id, buffer)
+
+    {[], %{state | live_queue: new_live_queue}}
+  end
+
+  @impl true
+  def handle_end_of_stream(Pad.ref(:input, pad_id), context, %{live_queue: live_queue} = state) do
```

Comment:
It's a good question but I don't think we should produce audio when we don't have any input pad unless we are explicitly told to

Reply:
I think we should consider a case where some room in Jellyfish is empty for a while, after there were some people

Reply:
In the case of hls_endpoint, we have silence_generator added as an input pad.
We can also make it stop the clock and remove live_queue instead of sending eof. In case of a new input pad, we would init live_queue and start the clock.
Also, we should take into consideration how the compositor handles those cases because those elements very often work together.
(compositor also sends eof after it gets eof on all inputs)